src/modules/App.jsx

```diff
@@ -0,0 +1,75 @@
+import React, { useState } from 'react';
+import Board from './Board';
+import Header from './Header';
+import Footer from './Footer';
+import PopUp from './PopUp';
+import { TransitionAlertsWin, TransitionAlertsLose } from './Alert';
+import { getInitialBoardArray, getValueFromLocalStorage } from './utils/helpers';
+
+const App = () => {
+  const [boardSize, setBoardSize] = useState(4);
+  const [bestScore, setBestScore] = useState(getValueFromLocalStorage('bestScore') || [0]);
+  const localHistory = getValueFromLocalStorage('2048-history');
+  const isLocalBoardSizeSame =
+    localHistory && localHistory[localHistory.length - 1].board.length === boardSize * boardSize;
+  const [history, setHistory] = useState(
+    localHistory && isLocalBoardSizeSame
```

Comment:
for 'long' logics I'd suggest using the function inside setState

Reply:
done, thanks.